elastic/kibana
# [Infra] Make nav react to Hosts view enabled flag changing

## Summary

Closes #140996

This PR adds an app updater and consumes the hosts view enabled setting via an Observable instead so that changes to this setting are reacted to by changing the nav and deep links in the Infra app plugin.

### How to test

- Open the Infrastructure app from the global nav
- See that Hosts is not visible in the Observability nav
- Go to Stack Management, Advanced Settings and find the Host views enabled setting and enable it, click Save
- Go back to Infrastructure app and observe that Hosts is now visible in the Observability nav without refreshing the page
